File: src/scitrader.marketdataserver/src/main/java/com/scitrader/marketdataserver/MarketDataServer.java
package com.scitrader.marketdataserver;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.scitrader.marketdataserver.exchange.bitmex.IBitmexWebsocketClient;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.Calendar;

import static spark.Spark.*;

@Singleton
public class MarketDataServer implements IMarketDataServer{

  Logger Log = LogManager.getLogger(MarketDataServer.class);
  private IBitmexWebsocketClient wsClient;

  @Inject
  public MarketDataServer(IBitmexWebsocketClient wsClient){

    this.wsClient = wsClient;
  }

  @Override
  public synchronized void Run() {

    // Setup Rest API endpoints
    get("/hello", (req, res)->{
      return "Affirmative Dave, the com.scitrader.marketdataserver is online";
    });

    this.wsClient.connect();

    // Wait
    while(true){
      //Log.info("server status ==> " + Calendar.getInstance().getTime());
      try {
        this.wait(2000);
      } catch (InterruptedException e) {

        e.printStackTrace();
      }
    }
  }
}

File: Kotlin/hello-kotlin-world/src/main/java/com/scitrader/marketdataserver/MarketDataServer.kt
package com.scitrader.marketdataserver

import com.google.inject.Inject
import com.scitrader.marketdataserver.exchange.bitmex.IBitmexWebsocketClient
import org.apache.logging.log4j.LogManager
import java.util.*

class MarketDataServer : IMarketDataServer {

    internal var Log = LogManager.getLogger(MarketDataServer::class.java)

    private var marketDataController: IMarketDataController
    private val syncObj = java.lang.Object()

    private var bitmexClient: IBitmexWebsocketClient

    @Inject
    constructor(bitmexClient : IBitmexWebsocketClient,
                marketDataController: IMarketDataController){

        this.marketDataController = marketDataController
        this.bitmexClient = bitmexClient;
    }

    @Synchronized override fun Run() {
        Log.info("The market data server is aliiiive!")

        this.marketDataController.Init()
        this.bitmexClient.connect()

        // Wait
        while (true) {
            Log.info("server status ==> " + Calendar.getInstance().getTime());
            try {
                synchronized(syncObj) {
                    syncObj.wait(10000)
                }
            } catch (e: InterruptedException) {

                e.printStackTrace()
            }

        }
    }
}